dir/path.go

```diff
@@ -1,88 +1,95 @@
+// package dir implements Notation directory structure.
+// [directory spec]: https://github.com/notaryproject/notation/blob/main/specs/directory.md
+//
+// Example:
+//
+//   - Read config.json:
+//     file, err := dir.ConfigFS().Open(dir.PathConfigFile)
+//
+//   - Get the path of config.json:
+//     path, err := dir.ConfigFS().SysPath(dir.PathConfigFile)
+//
+//   - Read trustpolicy.json:
+//     file, err := dir.ConfigFS().Open(dir.PathTrustPolicy)
+//
+//   - Get the path of trustpolicy.json:
+//     path, err := dir.ConfigFS().SysPath(dir.trustpolicy)
+//
+//   - Set custom configurations directory:
+//     dir.UserConfigDir = '/path/to/configurations/'
+//
+// Only user level directory is supported for RC.1, and system level directory
+// may be added later.
 package dir
 
 import (
-	"errors"
-	"io/fs"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+var (
+	UserConfigDir  string // Absolute path of user level {NOTATION_CONFIG}
+	UserLibexecDir string // Absolute path of user level {NOTATION_LIBEXEC}
 )
 
 const (
-	// ConfigFile is the name of config file
-	ConfigFile = "config.json"
+	// notation is the directory name for notation configurations.
+	notation = "notation"
+)
 
-	// LocalCertificateExtension defines the extension of the certificate files
+// The relative path to {NOTATION_CONFIG}
+const (
+	// PathConfigFile is the config.json file relative path.
+	PathConfigFile = "config.json"
+	// PathSigningKeys is the signingkeys file relative path.
+	PathSigningKeys = "signingkeys.json"
+	// PathTrustPolicy is the trust policy file relative path.
+	PathTrustPolicy = "trustpolicy.json"
+	// PathPlugins is the plugins directory relative path.
+	PathPlugins = "plugins"
+	// LocalKeysDir is the directory name for local key relative path.
+	LocalKeysDir = "localkeys"
+	// LocalCertificateExtension defines the extension of the certificate files.
 	LocalCertificateExtension = ".crt"
-
-	// LocalKeyExtension defines the extension of the key files
+	// LocalKeyExtension defines the extension of the key files.
 	LocalKeyExtension = ".key"
-
-	// LocalKeysDir is the directory name for local key store
-	LocalKeysDir = "localkeys"
-
-	// SignatureExtension defines the extension of the signature files
-	SignatureExtension = ".sig"
-
-	// SignatureStoreDirName is the name of the signature store directory
-	SignatureStoreDirName = "signatures"
-
-	// SigningKeysFile is the file name of signing key info
-	SigningKeysFile = "signingkeys.json"
-
-	// TrustPolicyFile is the file name of trust policy info
-	TrustPolicyFile = "trustpolicy.json"
-
-	// TrustStoreDir is the directory name of trust store
+	// TrustStoreDir is the directory name of trust store.
 	TrustStoreDir = "truststore"
 )
 
-// PathManager contains the union directory file system and methods
-// to access paths of notation
-type PathManager struct {
-	ConfigFS     UnionDirFS
-	LibexecFS    UnionDirFS
-	UserConfigFS UnionDirFS
+var userConfigDir = os.UserCacheDir
```

Comment:
Why are we defining userConfigDir variable?

Reply:
It's for unit test to replace the os.UserCacheDir() function with a mock function 😂

Reply:
Added coments.